pkg/resources/deployments/deployment.go

```diff
@@ -0,0 +1,235 @@
+
+import (
+	hawtiov1alpha1 "github.com/hawtio/hawtio-operator/pkg/apis/hawtio/v1alpha1"
+	"github.com/hawtio/hawtio-operator/pkg/resources/containers"
+	"github.com/hawtio/hawtio-operator/pkg/resources/environments"
+	"github.com/hawtio/hawtio-operator/pkg/resources/pods"
+	"github.com/hawtio/hawtio-operator/pkg/resources/volumes"
+	"github.com/hawtio/hawtio-operator/pkg/util/selectors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("package deployments")
+
+const (
+	serviceSigningSecretVolumeName         = "hawtio-online-tls-serving"
+	clientCertificateSecretVolumeName      = "hawtio-online-tls-proxying"
+	clientCertificateSecretVolumeMountPath = "/etc/tls/private/proxying"
+	deploymentRolloutAnnotation            = "hawtio.hawt.io/restartedAt"
+	hawtioVersionAnnotation                = "hawtio.hawt.io/hawtioversion"
+	hawtioTypeAnnotation                   = "hawtio.hawt.io/hawtioType"
+	configVersionAnnotation                = "hawtio.hawt.io/configversion"
+)
+
+// move this to util
+// Set labels in a map
+func labelsForInterconnect(name string) map[string]string {
+	return map[string]string{
+		selectors.LabelAppKey:      name,
+		selectors.LabelResourceKey: name,
+	}
+}
+
+// Create NewDeploymentForCR method to create deployment
+func NewDeploymentForCR(cr *hawtiov1alpha1.Hawtio, isOpenShift4 bool, openshiftVersion string, openshiftURL string, volumePath string, configResourceVersion string) *appsv1.Deployment {
+
+	reqLogger := log.WithName(cr.Name)
+	reqLogger.Info("Creating new Deployment for custom resource")
+
+	namespacedName := types.NamespacedName{
+		Name:      cr.Name,
+		Namespace: cr.Namespace,
+	}
+
+	annotations := map[string]string{
+		configVersionAnnotation: configResourceVersion,
+		hawtioTypeAnnotation:    cr.Spec.Type,
+		hawtioVersionAnnotation: cr.GetResourceVersion(),
+	}
+
+	dep, Spec := MakeDeployment(namespacedName, annotations, cr.Spec.Replicas, NewPodTemplateSpecForCR(cr, isOpenShift4, openshiftVersion, openshiftURL, volumePath))
+
+	dep.Spec = Spec
+
+	return dep
+
+}
+
+func MakeDeployment(namespacedName types.NamespacedName, annotations map[string]string, replicas int32, pts corev1.PodTemplateSpec) (deployment *appsv1.Deployment, spec appsv1.DeploymentSpec) {
+
+	labels := selectors.LabelsForHawtio(namespacedName.Name)
+	dep := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        namespacedName.Name,
+			Namespace:   namespacedName.Namespace,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+	}
+	Spec := appsv1.DeploymentSpec{
+		Replicas: &replicas,
+		Selector: &metav1.LabelSelector{
+			MatchLabels: labels,
+		},
+		Template: pts, //pods.NewPodTemplateSpecForCR(cr),
```

Comment:
Should that be replaced with the method call?

Reply:
kind of dependencies on parameters, a different list of parameters like  ```isOpenShift4, openshiftVersion, openshiftURL, volumePath``` so made function while calling `makeDeployment`